code/clue.js

```diff
@@ -1,59 +1,260 @@
   mystery.killer = randomSelector(suspects)
 
   // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
+  document.getElementById('killerText').style.background = mystery.killer.color
   document.getElementById(
-    'killerName'
+    'killerText'
   ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerImage').style.background = mystery.killer.image
+}
+
+// This function will be invoked when you click on the weapon card.
+const pickWeapon = () => {
+
+  mystery.weapon = randomSelector(weapons)
+
+  document.getElementById(
+    'weaponText'
+  ).innerHTML = `${mystery.weapon.name}`
+}
+
+// This function will be invoked when you click on the room card.
+const pickRoom = () => {
+
+  mystery.room = randomSelector(rooms)
+
+  document.getElementById(
+    'roomText'
+  ).innerHTML = `${mystery.room}`
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+document.getElementById('killer').addEventListener('click', pickKiller)
+document.getElementById('weapon').addEventListener('click', pickWeapon)
+document.getElementById('room').addEventListener('click', pickRoom)
+
 
 // STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
 // 'The murder was committed by Jacob Green, in the living room with a rope.'
+
+
+// Get modal element
+const modal = document.getElementById("simpleModal");
+
+// Get open modal button
+const modalBtn = document.getElementById("reveal");
+
+// Get close button
+const closeBtn = document.getElementById("closeBtn");
+
+// Function to open modal
+const reveal = () => {
+  modal.style.display = "flex";
+  modal.style.justifyContent = "center";
+  modal.style.alignItems = "center";
+  document.getElementById(
+    'revealText'
+  ).innerHTML = `Omg, the murder was committed by ${mystery.killer.firstName} ${mystery.killer.lastName}, in the ${mystery.room} with a ${mystery.weapon.name}.`
+
+}
+
+// Function to close modal
+const closeModal = () => {
+  modal.style.display = "none";
+}
+
+// Function to close modal if outside click/touch
+const outsideClick = (e) => {
+  if (e.target == modal) {
+    modal.style.display = "none";
+  }
+}
+
+// Listen for open click and open touch (on mobile devices)
+modalBtn.addEventListener("click", reveal);
+modalBtn.addEventListener("touchstart", reveal);
+
+// Listen for close click and close touch (on mobile devices)
+closeBtn.addEventListener("click", closeModal);
+closeBtn.addEventListener("touchstart", closeModal);
+
+// Listen for outside click and outside touch (on mobile devices)
+window.addEventListener("click", outsideClick);
+window.addEventListener("touchstart", outsideClick);
```

Comment:
Brilliant.  +1 for including both touch and click. 
I love details like this, great quality of life improvement being able to click outside of the modal.